x-pack/plugins/lens/public/editor_frame_service/editor_frame/config_panel/layer_panel.test.tsx

```diff
@@ -303,37 +303,83 @@ describe('LayerPanel', () => {
       expect(groups.findWhere((e) => e.prop('error') === '')).toHaveLength(1);
     });
 
-    it('should render the required warning when only one group is configured (with requiredMinDimensionCount)', async () => {
-      mockVisualization.getConfiguration.mockReturnValue({
-        groups: [
-          {
-            groupLabel: 'A',
-            groupId: 'a',
-            accessors: [{ columnId: 'x' }],
-            filterOperations: () => true,
-            supportsMoreColumns: false,
-            dataTestSubj: 'lnsGroup',
-          },
-          {
-            groupLabel: 'B',
-            groupId: 'b',
-            accessors: [{ columnId: 'y' }],
-            filterOperations: () => true,
-            supportsMoreColumns: true,
-            dataTestSubj: 'lnsGroup',
-            requiredMinDimensionCount: 2,
-          },
-        ],
-      });
-
-      const { instance } = await mountWithProvider(<LayerPanel {...getDefaultProps()} />);
+    it.each`
+      minDimensions | accessors | errors
+      ${1}          | ${0}      | ${1}
+      ${2}          | ${0}      | ${2}
+      ${2}          | ${1}      | ${2}
+    `(
+      'should render the required warning for $errors fields when only one group is configured with requiredMinDimensionCount: $minDimensions and $accessors accessors',
+      async ({ minDimensions, accessors, errors }) => {
+        mockVisualization.getConfiguration.mockReturnValue({
+          groups: [
+            {
+              groupLabel: 'A',
+              groupId: 'a',
+              accessors: [{ columnId: 'x' }],
+              filterOperations: () => true,
+              supportsMoreColumns: false,
+              dataTestSubj: 'lnsGroup',
+            },
+            {
+              groupLabel: 'B',
+              groupId: 'b',
+              accessors: [{ columnId: 'y' }].slice(0, accessors),
+              filterOperations: () => true,
+              supportsMoreColumns: true,
+              dataTestSubj: 'lnsGroup',
+              requiredMinDimensionCount: minDimensions,
+            },
+          ],
+        });
+        const { instance } = await mountWithProvider(<LayerPanel {...getDefaultProps()} />);
+
+        const errorMessage = errors === 1 ? 'Requires field' : 'Requires 2 fields';
+
+        const group = instance.find(EuiFormRow).findWhere((e) => e.prop('error') === errorMessage);
+
+        expect(group).toHaveLength(1);
+      }
+    );
+
+    it.each`
+      minDimensions | accessors
+      ${0}          | ${0}
+      ${0}          | ${1}
+      ${1}          | ${1}
+      ${1}          | ${2}
+      ${2}          | ${2}
+    `(
+      'should not render the required warning when only one group is configured with requiredMinDimensionCount: $minDimensions and $accessors accessors',
+      async ({ minDimensions, accessors, errors }) => {
```

Comment:
```suggestion
      async ({ minDimensions, accessors }) => {
```
forgot to remove `errors` here